apps/elasticsearch.go

```diff
@@ -5,8 +5,101 @@ import (
 
 	"github.com/GoogleCloudPlatform/ops-agent/confgenerator"
 	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/fluentbit"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/otel"
 )
 
+type MetricsReceiverElasticsearch struct {
+	confgenerator.ConfigComponent          `yaml:",inline"`
+	confgenerator.MetricsReceiverShared    `yaml:",inline"`
+	confgenerator.MetricsReceiverSharedTLS `yaml:",inline"`
+
+	Endpoint string `yaml:"endpoint" validate:"omitempty,url,startswith=http:|startswith=https:"`
+
+	Username string `yaml:"username" validate:"required_with=Password"`
+	Password string `yaml:"password" validate:"required_with=Username"`
+
+	CollectJVMMetrics     *bool `yaml:"collect_jvm_metrics" validate:"omitempty"`
+	CollectClusterMetrics *bool `yaml:"collect_cluster_metrics" validate:"omitempty"`
+}
+
+const (
+	defaultElasticsearchEndpoint = "http://localhost:9200"
+)
+
+var (
+	defaultElasticsearchCollectClusterMetrics = true
+)
+
+func (r MetricsReceiverElasticsearch) Type() string {
+	return "elasticsearch"
+}
+
+func (r MetricsReceiverElasticsearch) Pipelines() []otel.Pipeline {
+	if r.Endpoint == "" {
+		r.Endpoint = defaultElasticsearchEndpoint
+	}
+
+	if r.CollectClusterMetrics == nil {
```

Comment:
What if `CollectClusterMetrics` is not null and is set to "false" or "true". Looks like the behavior would be the same.

BTW, Apache and Tomcat has the same `collect_cluster_metrics` flag. Can we move the relevant logic like default value and field definition to a shared util (similar to `MetricsReceiverSharedTLS`, so that the code does not need to be duplicate and the behavior is consistent?)

Reply:
I refactored this into a shared struct; No other receiver uses it as far as I can tell (neither apache or tomcat are clustered), but it could come handy in the future if there are more receivers for cluster-based technologies set up like Elasticsearch.

Reply:
Great. Thanks!